tests/mesh/test_gradient.py

```diff
@@ -230,3 +238,332 @@ def test_grad_dxy_complex_shuffle():
     grad = df.gradient.gradient_of('fct')
 
     pd.testing.assert_frame_equal(grad, expected)
+
+# ---- gradient 2
+def test_gradient_3D_constant():
+    fkt = [5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5]
+    df = pd.DataFrame({'node_id': [1, 2, 2, 3, 4, 4, 5, 5, 5, 5, 6, 6, 7, 8, 8, 9],
+                        'element_id': [1, 1, 2, 2, 1, 3, 1, 2, 3, 4, 2, 4, 3, 3, 4, 4],
+                        'x': [0, 1, 1, 2, 0, 0, 1, 1, 1, 1, 2, 2, 0, 1, 1, 2],
+                        'y': [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2],
+                        'z': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        'fct': fkt}).set_index(['node_id', 'element_id'])
+
+    expected = pd.DataFrame({
+        'dfct_dx': np.zeros(9),
+        'dfct_dy': np.zeros(9),
+        'dfct_dz': np.zeros(9)
+    }, index=pd.RangeIndex(1, 10, name='node_id'))
+
+    grad = df.gradient_3D.gradient_of('fct').sort_index()
+
+    pd.testing.assert_frame_equal(grad.reset_index(), expected.reset_index(), check_dtype=False)
+
+
+def test_gradient_3D_dx():
```

Comment:
Please add a test with the index levels flipped. 

Like this one: https://github.com/boschresearch/pylife/blob/a9fe148ef1b152215a798f683608aca1f095c940/tests/mesh/test_gradient.py#L63